mkdocs/docs/backends/gpu-vulkan.md

```diff
@@ -1,10 +1,149 @@
-# Vulkan GPU backend
+# Vulkan GPU HAL Driver
 
-<!-- TODO(??): overview, when to use CPU vs GPU -->
+IREE can accelerate model execution on GPUs via [Vulkan][vulkan], a low-overhead
+graphics and compute API. Vulkan is cross-platform: it is available on many
+operating systems, including Android, Linux, and Windows. Vulkan is also
+cross-vendor: it is supported by most GPU vendors, including AMD, ARM, Intel,
+NVIDIA, and Qualcomm.
+
+<!-- TODO(??): when to use CPU vs GPU -->
+
+<!-- TODO(??): support matrix of different GPUs -->
 
 ## Prerequisites
 
-<!-- TODO(??): build from source, vcpkg -->
+In order to use Vulkan to drive the GPU, you need to have a functional Vulkan
+environment. IREE requires Vulkan 1.1 on Android and 1.2 elsewhere. It can be
+verified by the following steps:
+
+=== "Android"
+
+    Android mandates Vulkan 1.1 support since Android 10. You just need to
+    make sure the device's Android version is 10 or higher.
+
+=== "Linux"
+
+    Run the following command in a shell:
+
+    ``` shell
+    vulkaninfo | grep apiVersion
+    ```
+
+    If `vulkaninfo` does not exist, you will need to [install the latest Vulkan
+    SDK][vulkan-sdk]. For Ubuntu 18.04/20.04, installing via LuarG's package
+    repository is recommended, as it places Vulkan libraries and tools under
+    system paths so it's easy to discover.
+
+    If the showed version is lower than Vulkan 1.2, you will need to update the
+    driver for your GPU.
+
+
+=== "Windows"
+
+    Run the following command in a shell:
+
+    ``` shell
+    vulkaninfo | grep apiVersion
+    ```
+
+    If `vulkaninfo` does not exist, you will need to [install the latest Vulkan
+    SDK][vulkan-sdk].
+
+    If the showed version is lower than Vulkan 1.2, you will need to update the
+    driver for your GPU.
+
+
+## Get Runtime and Compiler
+
+### Get IREE runtime with Vulkan HAL driver
+
+Next you will need to get an IREE runtime that supports the Vulkan HAL driver
+so it can execute the model on GPU via Vulkan.
+
+<!-- TODO(??): vcpkg -->
+
+#### Build runtime from source
+
+Please make sure you have followed the [Getting started][get-started] page
+to build IREE for Linux/Windows and the [Android cross-compilation][android-cc]
+page for Android. The Vulkan HAL driver is compiled in by default on non-Apple
+platforms.
+
+<!-- TODO(??): a way to verify Vulkan is compiled in and supported -->
+
+If you want to explicitly specify HAL drivers to support, you will need to add
+`Vulkan` to the `IREE_HAL_DRIVERS_TO_BUILD` CMake list variable when
+configuring (for target).
+
+### Get compiler for SPIR-V exchange format
+
+Vulkan expects the program running on GPU to be expressed by the [SPIR-V][spirv]
+binary exchange format, which the model must be compiled into.
+
+<!-- TODO(??): vcpkg -->
+
+#### Build compiler from source
```

Comment:
Would be worth adding a section here for using the compiler _without building from source_, pointing at binary distributions and/or python packages.

One option would be to link to the frontend docs.

At least as a TODO for now, since some of that is still rough around the edges / incomplete (i.e. we don't have non-python binary releases).